api/envoy/api/v2/route/route.proto

```diff
@@ -325,6 +325,15 @@ message RouteMatch {
   message GrpcRouteMatchOptions {
   }
 
+  message CredentialMatchOptions {
+    // If specified, the route will match against whether or not a certificate is presented.
+    google.protobuf.BoolValue presented = 1;
+
+    // If specified, the route will match against whether or not the client certificate has expired.
+    // If no certificate presented, it is treated as not expired.
+    google.protobuf.BoolValue expired = 2;
```

Comment:
Do we ever want to support this?

Reply:
I imagine this could be useful if 'allow_expired_certificate' is enabled, then can route to a service that performs cert renewals or notifications etc.. It's similar to our #8248 PR - to be able to route based on whether a cert was validated or not.

Reply:
Ack, this makes sense.